src/harness/BenchmarkDotNet.Extensions/PartitionFilter.cs

```diff
@@ -0,0 +1,32 @@
+using BenchmarkDotNet.Filters;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using BenchmarkDotNet.Running;
+
+
+public class PartitionFilter : IFilter
+{
+    private readonly int? _partitionsCount;
+    private readonly int? _partitionIndex; // indexed from 0
+    private int _counter = 0;
+ 
+    public PartitionFilter()
+    {
+        _partitionsCount = ReadFromEnvironmentVariable("PARTITIONS_COUNT");
+        _partitionIndex = ReadFromEnvironmentVariable("PARTITION_INDEX");
+    }
+ 
+    public bool Predicate(BenchmarkCase benchmarkCase)
+    {
+        if (!_partitionsCount.HasValue || !_partitionIndex.HasValue)
```

Comment:
nit: this can just check `partitionsCount` since if you didn't specify that it doesn't matter.
```suggestion
        if (!_partitionsCount.HasValue)
```

Reply:
But if you didn't specify which partition you want, how do you know which one to run? Do we default to 0?

Reply:
Sorry. Half a thought. I'd do that as part of argument validation instead of silently fail here. Specifying one without the other isn't a valid state.

Reply:
done.